### PR TITLE
Avoid losing errors from the underlying Windows functions.

### PR DESCRIPTION
**Pull request by keithbostic**
@markbenvenuto, can you review & merge if this is correct?

### CODE REVIEW

File: src/os_win/os_mtx_cond.c [hunk 1]
```diff
@@ -94,18 +94,14 @@ __wt_cond_wait_signal(
...
-	}
+	if (ret == 0 && GetLastError() == ERROR_TIMEOUT)
+		*signalled = 0;
 
 	(void)__wt_atomic_subi32(&cond->waiters, 1);
 
 	if (locked)
 		LeaveCriticalSection(&cond->mtx);
-	if (ret != 0)
+	if (ret == 0)
```
Comment (markbenvenuto) [inline src/os_win/os_mtx_cond.c]: `SleepConditionVariableCS` returns non-zero on success. See https://msdn.microsoft.com/en-us/library/windows/desktop/ms686301(v=vs.85).aspx

**Comment by keithbostic** [comment]
@markbenvenuto, I tried again, please let me know what you think.

**Comment by markbenvenuto** [comment]
@keithbostic The control flow is not a little ugly on the error case, but I wanted to be sure we set the state correctly on error. Let me know what you think.

**Comment by keithbostic** [comment]
Thanks, @markbenvenuto, looks great!

**Comment by keithbostic** [comment]
Reference WT-2090.

**Pull request merged**